Backend/src/main/java/com/project/omega/service/implmentations/OrderServiceImpl.java

```diff
@@ -29,12 +33,12 @@ public Order createOrder(Order order) {
     }
 
     @Override
-    public Order getOrderById(Long id) throws OrderNotFoundException {
-        Optional<Order> order = orderRepository.findById(id);
-        if (!order.isPresent()) {
-            throw new OrderNotFoundException("Order with id" + id + " not found");
-        }
-        return order.get();
+    public Order getOrderById(Long id) throws NoRecordsFoundException {
+            Optional<Order> order = orderRepository.findById(id);
+            if (!order.isPresent()){
+                throw new NoRecordsFoundException(messages.getMessage("message.orderNotFound" , null, null));

```

Comment:
The previous exception (OrderNotFoundException) is the correct one to use here and in updateOrderById. Change to that please, and you should be good to go.